File: src/main/java/fr/lewon/web/bot/entities/input/mission/ActionMission.java
package fr.lewon.web.bot.entities.input.mission;

import fr.lewon.bot.http.body.urlencoded.FUEMember;
import fr.lewon.web.bot.entities.input.Action;

public abstract class ActionMission extends Action {

	@FUEMember("id_mission")
	private Long idMission;

	@FUEMember("id_member_mission")
	private Long idMemberMission;

	public ActionMission(String action, Long idMission, Long idMemberMission) {
		super("Missions", action);
		this.idMission = idMission;
		this.idMemberMission = idMemberMission;
	}

}

File: src/main/kotlin/fr/lewon/bot/hh/entities/input/mission/ActionMission.kt
package fr.lewon.bot.hh.entities.input.mission

import com.fasterxml.jackson.annotation.JsonProperty
import fr.lewon.bot.hh.entities.input.Action

abstract class ActionMission(action: String, idMission: Long?, idMemberMission: Long?) : Action("Missions", action) {

    @JsonProperty("id_mission")
    val idMission: Long? = idMission
    @JsonProperty("id_member_mission")
    val idMemberMission: Long? = idMemberMission
}